test/receiver/receiver.go

```diff
@@ -65,13 +66,44 @@ func main() {
 	outDir := flag.String("outdir", "", "Output directory. Absolute path. Optional.")
 	profiler := flag.String("profiler", "", "Where should the profiler listen?")
 	exitAfter := flag.Duration("exitAfter", time.Second*10, "Exit after not receiving any message for this time. Will work only of outDir == ''. Will not exit if Duration is 0")
+	localAPIPort := flag.Int64("local-api-port", 0, "specify which port the local HTTP API should be listening on")
 
 	flag.Parse()
 
 	if *portsStr == "" {
 		log.Fatal("please supply the ports argument")
 	}
 
+	var currentStatus = &struct {
+		sync.Mutex
+		Ready                  bool
+		dataProcessed          bool
+		timestampLastProcessed time.Time
+		IdleTimeMilliSecs      int64
+	}{sync.Mutex{}, false, false, time.Now(), 0}
+
+	http.HandleFunc("/status", func(w http.ResponseWriter, req *http.Request) {
+		currentStatus.Lock()
+		defer currentStatus.Unlock()
+		if currentStatus.dataProcessed {
+			currentStatus.IdleTimeMilliSecs = time.Since(currentStatus.timestampLastProcessed).Milliseconds()
+		}
+		data, err := json.Marshal(currentStatus)
+		if err != nil {
+			log.Printf("error when json marshaling status: %v", currentStatus)
+		}
+		fmt.Fprint(w, string(data))
+	})
+	go func() {
+		log.Printf("local API setup open port %d", *localAPIPort)
```

Comment:
When port flag is 0, this will log opening port 0 when in fact no port will be open.